src/snowflake/connector/connection.py

```diff
@@ -349,6 +365,17 @@ def __init__(
                     f" known ones are {list(connections.keys())}"
                 )
             kwargs = {**connections[connection_name], **kwargs}
+        elif empty_kwargs:
+            # connection_name is None and kwargs was empty when called
+            def_connection_name = CONFIG_MANAGER["default_connection_name"]
+            connections = CONFIG_MANAGER["connections"]
+            if def_connection_name not in connections:
+                raise Error(
+                    f"Default connection with name '{def_connection_name}' "
+                    "cannot be found, known ones are "
+                    f"{list(connections.keys())}"
```

Comment:
You don't need `.keys()` here since iteration over a dictionary is always by key by default.  Also, I'm not sure the list view will be super helpful.  Yeah, okay, it's just in an error message, but something like this might read better:

```python
COMMASPACE = ', '
...
f"{COMMASPACE.join(connections)}"
```

Reply:
Sorry, that is exactly how I leaked the testing account password last time, when the error a few lines above was introduced. On some Python version and OS what you suggest also inserted the values and not only the keys!